### ISSUE TITLE
Reporting for Cost Planning and Adjusting: Details on usage for ope classes Spring 2024 to verify or correct the cost allocation and calculation

### ISSUE THREAD

**Issue by schwesig** [nerc-project/operations#627]
## to set these 3 issues in perspective
- https://github.com/nerc-project/operations/issues/258
  - focus on Professors & TA's, to better analyze the progress of a student or class
- https://github.com/nerc-project/operations/issues/374
  - focus on Accounting and creating invoices
- https://github.com/nerc-project/operations/issues/627
  - focus on the usage of resources to adjust and check cost calculations for upcoming classes & price tags

## Details
- Focus: Cost allocation & planning, check previous usage, and adjust calculation/price tags

We need to verify the actual/real usage for the OPE classes for the Spring 2024 semester. 
This helps to verify the calculated costs we made.

- A one-time report is the minimum.
- A general report function (dashboard?) would be nice to have now, but it is definitely a must in the future.

/CC @msdisme

**Comment by schwesig** [comment]
Status:
the timeout between pods messed up some of the retention times and data.
working on getting the data from last classes by
- collecting infra
- collecting obs
- collecting prod
- checking alerts and grafana
- getting Naved's backup from invoicing

**Comment by msdisme** [comment]
@schwesig any update on the usage costs per class?

**Comment by schwesig** [comment]
> @schwesig any update on the usage costs per class?
@msdisme 
waiting on some last input, plan to finish results tomorrow (Thur), latest end of week (Fri)

**Comment by schwesig** [comment]
We had a call today
@msdisme @naved001 @schwesig 
and figured out a calculations
details are finalized internally

**Comment by msdisme** [comment]
@DanNiESh @computate On this [spreadsheet](https://docs.google.com/spreadsheets/d/1tIVlFx7pyVy55K_WyCWhSyNNoB-nn7ZeY9Am42csgyg/edit?usp=sharing) I have the cost for the semester of usage of RHODS by cs506, cs210 and ec440.  Do we have any data that would show the amount of usage by each class?

**Comment by DanNiESh** [comment]
> @DanNiESh @computate On this [spreadsheet](https://docs.google.com/spreadsheets/d/1tIVlFx7pyVy55K_WyCWhSyNNoB-nn7ZeY9Am42csgyg/edit?usp=sharing) I have the cost for the semester of usage of RHODS by cs506, cs210 and ec440. Do we have any data that would show the amount of usage by each class?

We don't have any data that shows the usage by each class AFAIK.

**Comment by computate** [comment]
> @DanNiESh @computate On this [spreadsheet](https://docs.google.com/spreadsheets/d/1tIVlFx7pyVy55K_WyCWhSyNNoB-nn7ZeY9Am42csgyg/edit?usp=sharing) I have the cost for the semester of usage of RHODS by cs506, cs210 and ec440. Do we have any data that would show the amount of usage by each class?

I'm not aware of metrics that would allow us to gather metrics per class, since classes are not separated by namespace, or even by labels that I'm aware of. There are some possible metrics that could be used that report the CPU, memory, and GPU metrics per student by pod name, but not by class: 

- container_cpu_usage_seconds_total{namespace="rhods-notebooks", pod="..."}
- container_memory_usage_bytes{namespace="rhods-notebooks", pod="..."}
- DCGM_FI_DEV_GPU_UTIL{cluster="nerc-ocp-prod",exported_namespace="rhods-notebooks",exported_pod="..."}
